week2/homework/src/index.js

```diff
@@ -1,3 +1,68 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+
+const { readFile, writeFile } = require('fs');
+
+const { promisify } = require('util');
+
+
+//const appendFileWithPromise = promisify(appendFile);
+const readFileWithPromise = promisify(readFile);
+const writeFileWithPromise = promisify(writeFile);
+
+const TODO_FILE = 'todo.json';
+
+async function main() {
+  const [, , cmd, ...args] = process.argv;
+
+  switch (cmd) {
+    case 'add': {
+      const data = await readFileWithPromise(TODO_FILE, 'utf8').catch(() => '[]');
+      const todos = JSON.parse(data);
+      const newTodo = args.join(' ');
+      todos.push(newTodo);
+      await writeFileWithPromise(TODO_FILE, JSON.stringify(todos));
+      break;
+    }
+    case 'list': {
+      const data = await readFileWithPromise(TODO_FILE, 'utf8').catch(() => '[]');
+      const todos = JSON.parse(data);
+      console.info(todos);
+      break;
+    }
+    case 'reset': {
+      await writeFileWithPromise(TODO_FILE, JSON.stringify([]));
+      const data = await readFileWithPromise(TODO_FILE, 'utf8').catch(() => '[]');
+      break;
+    }
+    case 'remove': {
+      const data = await readFileWithPromise(TODO_FILE, 'utf8').catch(() => '[]');
+      const todos = JSON.parse(data);
+      const removeTodo = args.join(' ');
+      todos.slice(removeTodo);
+      await writeFileWithPromise(TODO_FILE, (JSON.stringify(todos)));
+      break;
+    }
+    case 'update': {
+      const data = await readFileWithPromise(TODO_FILE, 'utf8').catch(() => '[]');
+      const todos = JSON.parse(data);
+      const updatedTodo = args.join(' ');
+      todos.splice(updatedTodo);
```

Comment:
Kind of the same problem as above, but also: `splice` should have 3 arguments here (you should read the [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)): 
1. 1st argument is the index of the array where you want your new element to be; 
2. 2nd argument is the number of elements you want to remove from the array; 
3. 3rd argument is the value you want to insert. 

So we want our element to be inserted at the provided `index`. That will be our first argument. Then we want to remove it, so the 2nd argument will be `1`. And then we'll want to insert the value that's in the args. 

I've already given you a lot of info, let me know if it is still unclear for you.